src/main/java/bio/terra/common/LockOperation.java

```diff
@@ -27,20 +28,55 @@ public enum LockOperation {
    * @return additional guidance for users investigating a failed lock operation, usually as part of
    *     a failed job / flight.
    */
-  public List<String> getErrorDetails() {
+  public List<String> getErrorDetails(ResourceLocks conflictingLocks) {
     var template =
         "A failure to obtain %s on a resource likely means that it's already %s by another process.";
     if (description != null && conflict != null) {
+      if (conflictingLocks != null && !isEmpty(conflictingLocks)) {
+        template =
+            template
+                + " Conflicting lock(s): %s. You can remove a lock by using the unlock API endpoint with the ID of the resource (Snapshot or Dataset), the name of the lock, and forceUnlock set to false.";
+        if (this == LOCK_EXCLUSIVE) {
+          // include any locks that exist both exclusive and shared (but only one will be populated)
+          if (isExclusive(conflictingLocks)) {
+            return List.of(
+                template.formatted(description, conflict, conflictingLocks.getExclusive()));
+          } else if (areShared(conflictingLocks)) {
+            String locks = String.join(", ", conflictingLocks.getShared());
+            return List.of(template.formatted(description, conflict, locks));
+          }
```

Comment:
For the case where we're trying to obtain an exclusive lock, why not return the list of exclusive and shared locks? I think they would have to unlock all of them in order to obtain the lock. (But, it's been a while since I've looked at the locking/unlocking functionality.)

Reply:
You're right they all need to be removed, but there will only ever exist shared locks or an exclusive lock. Thats why I set it up this way. But, maybe it would just be simpler to join them all even though they won't both exist. what do you think?